com.ca.lsp.cobol/lsp-service-cobol/src/main/java/com/ca/lsp/cobol/service/MyTextDocumentService.java

```diff
@@ -195,7 +198,7 @@ public void didSave(DidSaveTextDocumentParams params) {
 
   @Override
   public void observerCallback(@Nonnull RunAnalysisEvent event) {
-    docs.forEach((key, value) -> analyzeChanges(key, value.getText()));
+    docs.forEach((key, value) -> analyzeDocumentFirstTime(key, value.getText()));
```

Comment:
what about to introduce a small Facade that accept the mode (DID_OPEN or DID_CHANGE) and invoke the correct implementation? The thing that ```analyzeDocumentFirstTime``` could be executed when we need to use DID_OPEN mode could not be clear for the developer

Reply:
In fact, we just need to create a more descriptive name for these methods. Something like `analyzeCompleteDocument` and `analyzeIncompleteDocument`